worlds/minecraft/__init__.py

```diff
@@ -96,66 +94,63 @@ def _get_mc_data(self):
+        item_class = ItemClassification.filler
+        if name in Constants.item_info["progression_items"]:
+            item_class = ItemClassification.progression
+        elif name in Constants.item_info["useful_items"]:
+            item_class = ItemClassification.useful
+        elif name in Constants.item_info["trap_items"]:
+            item_class = ItemClassification.trap
+
+        return MinecraftItem(name, item_class, self.item_name_to_id.get(name, None), self.player)
+
+    def create_event(self, region_name, event_name):
+        region = self.multiworld.get_region(region_name, self.player)
+        loc = MinecraftLocation(self.player, event_name, None, region)
+        loc.place_locked_item(self.create_event_item(event_name))
+        region.locations.append(loc)
+
+    def create_event_item(self, name):
+        item = self.create_item(name)
+        item.classification = ItemClassification.progression
+        return item
 
     def create_regions(self):
-        def MCRegion(region_name: str, exits=[]):
-            ret = Region(region_name, self.player, self.multiworld)
-            ret.locations = [MinecraftAdvancement(self.player, loc_name, loc_data.id, ret)
-                for loc_name, loc_data in advancement_table.items()
-                if loc_data.region == region_name]
-            for exit in exits:
-                ret.exits.append(Entrance(self.player, exit, ret))
-            return ret
-
-        self.multiworld.regions += [MCRegion(*r) for r in mc_regions]
-        link_minecraft_structures(self.multiworld, self.player)
+        # Create regions
+        for region_name, exits in Constants.region_info["regions"]:
+            r = Region(region_name, self.player, self.multiworld)
+            for exit_name in exits:
+                r.exits.append(Entrance(self.player, exit_name, r))
+            self.multiworld.regions.append(r)
+
+        # Bind mandatory connections
+        for entr_name, region_name in Constants.region_info["mandatory_connections"]:
+            e = self.multiworld.get_entrance(entr_name, self.player)
+            r = self.multiworld.get_region(region_name, self.player)
+            e.connect(r)
+
+        # Add locations
+        for region_name, locations in Constants.location_info["locations_by_region"].items():
+            region = self.multiworld.get_region(region_name, self.player)
+            for loc_name in locations:
+                loc = MinecraftLocation(self.player, loc_name,
+                    self.location_name_to_id.get(loc_name, None), region)
+                region.locations.append(loc)
+
+        # Add events
+        self.create_event("Nether Fortress", "Blaze Rods")
+        self.create_event("The End", "Ender Dragon")
+        self.create_event("Nether Fortress", "Wither")
+
+        # Shuffle the connections
+        shuffle_structures(self)
+
+    def create_items(self):
+        self.multiworld.itempool += build_item_pool(self)
+
+    def set_rules(self):
```

Comment:
While I have no issue with this, `set_rules = set_rules` is valid if you prefer.